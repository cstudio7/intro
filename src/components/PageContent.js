import React, { Component } from 'react';
import {
    StyleSheet,
    Dimensions,
    Animated,
    Easing,
    Image,
    View,
    Text,
    ScrollView
} from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { setIndex } from '../actions/PageActions';

const { height: deviceHeight, width: deviceWidth } = Dimensions.get('window');

class PageContent extends Component {
    constructor(props) {
        super(props);
        this.ref = React.createRef();
    }

    componentDidUpdate(prevProps, prevState) {
        if (prevProps.offsetX !== this.props.offsetX) {
            this.ref.scrollTo({ x: this.props.offsetX, y:0, animated: true });
        }
    }

    scrollEvent = (event) => {
        // Get the index which will equal the page index showing
        const currentIndex = Math.round(event.nativeEvent.contentOffset.x / event.nativeEvent.layoutMeasurement.width);
        this.props.setIndex(currentIndex);
    }

    renderPages() {
        return this.props.pages.map((page, index) =>
            <View
                key={index}
                style={styles.pageContainer}
            >
                <Animated.View>
                    <Image
                        style={styles.image}
                        source={{uri: page.image}}
                    />
                </Animated.View>
                <View style={styles.titleView}>
                    <Text style={[styles.titleText, {color: this.props.lightBackground ? '#000' : '#FAFAFA'}]}>
                        {page.title}
                    </Text>
                </View>
                <View>
                    <Text style={[styles.subtitleText, {color: this.props.lightBackground ? '#000' : '#FAFAFA'}]}>
                        {page.subtitle}
                    </Text>
                </View>
            </View>
        );
    }

    render() {
        return (
            <ScrollView
                ref={(ref) => this.ref = ref}
                horizontal
                pagingEnabled={true}
                onMomentumScrollEnd={this.scrollEvent} //triggers when scroll is ending
                scrollEventThrottle={100} //limits the time events fire
                showsHorizontalScrollIndicator={false}
                contentContainerStyle={styles.container}
            >
                {this.renderPages()}
            </ScrollView>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        padding: 10,
        alignItems: 'center',
    },
    pageContainer: {
        width: deviceWidth,
        alignItems: 'center'
    },
    image: {
        width: 200,
        height: 200,
        resizeMode: 'cover',
        alignSelf: 'center'
    },
    titleText: {
        fontSize: 24,
        fontWeight: 'bold',
        textAlign: 'center',
        padding: 5
    },
    subtitleText: {
        fontSize: 16,
        fontWeight: 'normal',
        textAlign: 'center',
        padding: 0,
    },
    titleView: {
        marginTop: 16
    }
})

const mapStateToProps = (state) => {
    return {
        currentIndex: state.page.currentIndex
    }
}

PageContent.propTypes = {
    pages: PropTypes.arrayOf(PropTypes.shape({
        backgroundColor: PropTypes.string.isRequired,
        image: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        subtitle: PropTypes.string.isRequired
    })).isRequired,

    offsetX: PropTypes.number.isRequired,
    lightBackground: PropTypes.bool
}


export default connect(
    mapStateToProps, {
    setIndex
})(PageContent);
