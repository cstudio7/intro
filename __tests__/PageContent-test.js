import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import PageContent from '../src/components/PageContent.js';



Enzyme.configure({ adapter: new Adapter() });


const mockStore = configureMockStore()


describe('PageContent wrapped with redux is ok', () => {
    let wrapper, store;

    const initialState = {currentIndex: 0}

    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = shallow(<PageContent store={store} />);
    });

    it('renders alright', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('gets redux props', () => {
        expect(wrapper.props().currentIndex).toEqual(0)
    });

});
