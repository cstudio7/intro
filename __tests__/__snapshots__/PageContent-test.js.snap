// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent wrapped with redux is ok renders alright 1`] = `undefined`;
